### PR TITLE
Add ssl argument to Mongo connection

### PR DESCRIPTION
**Pull request by babs**
### What does this PR do?
Add ssl argument to Mongo connection

### CODE REVIEW

File: salt/states/mongodb_user.py [hunk 1]
```diff
@@ -58,6 +59,9 @@ def present(
     authdb
         The database in which to authenticate
 
+    ssl
+        Use ssl/tls to connect
+
```
Comment (waynew) [inline salt/states/mongodb_user.py:64]: **suggestion** add versionadded, e.g.

```suggestion
    ssl
        Use ssl/tls to connect

        .. versionadded:: Phosphorus

```

**Review by waynew — CHANGES_REQUESTED** [review]
I really like this addition, for obvious reasons :smile: 

We *do* require test coverage for code that's changed, and we don't really have that here. Is that something that you'd be wiling to add?

If not, no big deal - we'll just add the help-wanted label and will be able to merge when we can write tests for it.

Also this will need a [changelog entry](https://docs.saltstack.com/en/latest/topics/development/changelog.html)

**Comment by babs** [comment]
I'll call for help on that :/

**Comment by waynew** [comment]
I went ahead and implemented the changes+tests. Just re-merged the latest changes from master in, so :crossed_fingers: hopefully all the tests pass now.

**Comment by babs** [comment]
@waynew done :)
sorry for the delay.

**Comment by dwoz** [comment]
Closing this due to inactivity. Anyone should feel free to re-open it if they want to see it through to the end in one release cycle.

**Comment by babs** [comment]
It's still relevant unless another PR has been merged that adds ssl to mongodb...

**Comment by dwoz** [comment]
@babs Please rebase and fix the conflicts.

**Comment by babs** [comment]
@waynew might need a second pair of eyes and possibly hands to check the rebase :)

**Comment by babs** [comment]
@waynew might need your help on mongo ssl tests

**Comment by babs** [comment]
@dwoz do you know somebody that could give me a hand on fixing the instace tests ?

**Comment by babs** [comment]
@dwoz do you know somebody that could give me a hand on fixing the instance tests ?

**Comment by babs** [comment]
re-run all

**Comment by babs** [comment]
re-run all

**Review by dwoz — APPROVED** [review]

**Pull request merged**